js/coinsbit.js

```diff
@@ -0,0 +1,742 @@
+                //             limit: 2,
+                //             total: 334,
+                //             orders: [
+                //                 {
+                //                     id: 8620142523,
+                //                     left: "1.56772",
+                //                     market: "BTC_USDT",
+                //                     amount: "1.56772",
+                //                     type: "limit",
+                //                     price: "46752.63032535",
+                //                     timestamp: 1640031161.544,
+                //                     side: "buy",
+                //                     dealFee: "0",
+                //                     takerFee: "0",
+                //                     makerFee: "0",
+                //                     dealStock: "0",
+                //                     dealMoney: "0"
+                //                 },
+                //                 ...
+                //             ]
+                //         },
+                //         code: 200
+                //     }
+                const resultBids = this.safeValue (responses[0], 'result', {});
+                const resultAsks = this.safeValue (responses[1], 'result', {});
+                const bids = this.safeValue (resultBids, 'orders', []);
+                const asks = this.safeValue (resultAsks, 'orders', []);
+                const response = { 'bids': bids, 'asks': asks };
+                const priceKey = 'price';
+                const amountKey = 'amount';
+                const timestamp = this.milliseconds ();
+                orderbook = this.parseOrderBook (response, symbol, timestamp, 'bids', 'asks', priceKey, amountKey);
+            }
+        }
+        // orderbook['nonce']  <<< TODO >>> -- atm I dont know how to handle that, as the response doesn't have right that property
+        return orderbook;
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        const request = this.handleMarketTypeAndParams ('fetchOHLCV', undefined, params)[1];
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        request['market'] = market['id'];
+        const duration = this.parseTimeframe (timeframe);
+        const maxAllowedDiapason = this.options['maxDiapasonsForTimeframes'][timeframe];
+        let maxRequestedMilliSeconds = null;
+        if (limit === undefined) {
+            maxRequestedMilliSeconds = maxAllowedDiapason * 1000;
+        } else {
+            maxRequestedMilliSeconds = Math.min (this.sum (limit, 1) * duration, maxAllowedDiapason) * 1000;
+        }
+        let response = null;
+        if (market['spot']) {
+            request['interval'] = this.timeframes[timeframe];
+            const now = this.milliseconds ();
+            let startTime = since;
+            let endTime = now + 1000; // At first, Let's add one second
```

Comment:
This line is a violation of this rule: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#adding-numbers-and-concatenating-strings

Reply:
Should I use `... = Precise.stringAdd (now.ToString (), '1000')`?

Reply:
No, `Precise` is only used for fees, amounts, prices, costs, precisions and monetary values.
The timestamp arithmetic is always integer-based, therefore you should use `this.sum (now, 1000)`. We cannot use `+` here, it's for strings only.

Reply:
ahh. done.